### PR TITLE
Issues with NWM nc files during subsetting 

### PR DESCRIPTION
**Pull request by jzwart**
copied from other PR: We were getting weird values when reading the streamflow data from the subsetted nc files. We looked back to the subset_nwm.R script and think we fixed part of the problem. The streamflow dimensions were being screwed up when put into the new_nc files. We now loop through the sites (and ref time if forecast) and use ncvar_put to insert streamflow data into new_nc as well as multiply by the scale factor. If not multiplied by the scale factor when data is put into the nc, the data read out will be 100x's too small. Double check our looping - we think we mirrored what you had but couldn't test it.

### CODE REVIEW

**Comment by jzwart** [comment]
Sorry that wasn't a great description from the last pull request. When running 
 ```ncvar_put(new_nc, new_nc$var$streamflow, streamflow)``` 
we think it's writing data from streamflow matrix like you would write a book in English (left to right, top to bottom), but then it is reading the data like a table. So the first site has data from `streamflow[1:(nrow(streamflow) / ncol(streamflow)), ]`, the second site has data from `streamflow[(nrow(streamflow) / ncol(streamflow)):((nrow(streamflow) / ncol(streamflow))*2), ]`, and so on. I added some code to try and show our issues

File: recreate_nc_issues.R [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+                    rep(rep(0,n),new_nc$dim$feature_id$len - 3)),
+                    ncol= new_nc$dim$feature_id$len)
+
+ylim = range(streamflow)
+col= grDevices::rainbow(n = ncol(streamflow))
+plot(streamflow[,1],cex=0,ylim=ylim)
+for(i in 1:ncol(streamflow)){
+  lines(streamflow[,i],col=col[i],lwd=3)
+}
+
```
Comment (jzwart) [inline recreate_nc_issues.R]: image

File: recreate_nc_issues.R [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+                              count = c(1, -1))
+}
+
+ylim = range(streamflow_nc)
+col= grDevices::rainbow(n = ncol(streamflow_nc))
+plot(streamflow_nc[,1],cex=0,ylim=ylim)
+for(i in 1:ncol(streamflow_nc)){
+  lines(streamflow_nc[,i],col=col[i],lwd=3)
+} #  they're all the same 'hydrograph'
+
```
Comment (jzwart) [inline recreate_nc_issues.R]: image

File: 1_data/src/subset_nwm.R [hunk 3]
```diff
@@ -99,8 +99,28 @@ subset_nwm <- function(ind_file, model_configuration, comids, remake_file, gd_co
...
+      for(r in 1:nc$dim$reference_time$len) {
+        ncvar_put(new_nc,
+                  new_nc$var$streamflow,
+                  streamflow[,s]*(1/new_nc$var$streamflow$scaleFact),  # need to multiply streamflow data by scale factor in nc var
+                  start = c(s, 1, r),
+                  count = c(1,-1, 1))
+      }
+    }
+  }
+
```
Comment (dblodgett-usgs) [inline 1_data/src/subset_nwm.R]: I'm implementing this a little more directly. We don't really need the streamflow array at all.

Comment (aappling-usgs) [reply 1_data/src/subset_nwm.R]: :+1:
and thanks for giving up your vacation time for this! it really is OK if you need to wait until next week, but we'll gladly use whatever you're able to work up sooner

Comment (dblodgett-usgs) [reply 1_data/src/subset_nwm.R]: With what I'm implementing locally, I now get the following for nc (from thredds) and new_nc (from a local file):
`plot(ncvar_get(nc, "streamflow",start=c(site_inds[1],1), count=c(1,-1)))`
screen shot 2018-05-09 at 3 51 45 pm
`plot(ncvar_get(new_nc, "streamflow",start=c(1,1), count=c(1,-1)))`
screen shot 2018-05-09 at 3 51 22 pm

Comment (aappling-usgs) [reply 1_data/src/subset_nwm.R]: looks great!

Comment (dblodgett-usgs) [reply 1_data/src/subset_nwm.R]: To be thorough... 
`plot(ncvar_get(nc,"streamflow",start=c(site_inds[1],1,1), count=c(1,-1,1)))`
screen shot 2018-05-09 at 4 02 59 pm

`plot(ncvar_get(new_nc,"streamflow",start=c(1,1,1), count=c(1,-1,1)))`
screen shot 2018-05-09 at 4 03 25 pm

Comment (jzwart) [reply 1_data/src/subset_nwm.R]: awesome!

Comment (dblodgett-usgs) [reply 1_data/src/subset_nwm.R]: Should have done this the first time around... sorry, I was in a rush and not going through the right checks. I'm rebuilding all the netcdf files now and I think you should have usable output.

**Comment by dblodgett-usgs** [comment]
I think we can close this. I'll put in a new one in a minute. Not really clear to me what all I should be checking in. I'm getting a lot of rebuilds and different hashes... Will start with a few obvious things and you can tell me if I should add more.

**Comment by aappling-usgs** [comment]
Sounds good! You should git commit any new/changed files in the `build` folder, `.ind` files, and code and remake files. Only thing to be careful not to commit is data files, but hopefully our .gitignore file is already doing a lot of good ignoring for us on that.

**Comment by aappling-usgs** [comment]
closing at @dblodgett-usgs 's request

**Pull request closed without merging**